src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,652 @@
+            sage: A(L).affine_fundamental_group()
+            Finitely presented group
+            < x0, x1, x2, x3, x4 | x4*x0*x4^-1*x0^-1,
+                                   x0*x2*x3*x2^-1*x0^-1*x3^-1,
+                                   x1*x2*x4*x2^-1*x1^-1*x4^-1,
+                                   x2*x3*x0*x2^-1*x0^-1*x3^-1,
+                                   x2*x4*x1*x2^-1*x1^-1*x4^-1,
+                                   x4*x1*x4^-1*x3^-1*x2^-1*x1^-1*x2*x3 >
+            sage: H = A(x, y, x + y)
+            sage: H.affine_fundamental_group()
+            Finitely presented group
+            < x0, x1, x2 | x0*x1*x2*x1^-1*x0^-1*x2^-1, x1*x2*x0*x1^-1*x0^-1*x2^-1 >
+            sage: H.affine_fundamental_group()  # repeat to use the attribute
+            Finitely presented group
+            < x0, x1, x2 | x0*x1*x2*x1^-1*x0^-1*x2^-1, x1*x2*x0*x1^-1*x0^-1*x2^-1 >
+            sage: T.<t> = QQ[]
+            sage: K.<a> = NumberField(t^3 + t + 1)
+            sage: L.<x, y> = OrderedHyperplaneArrangements(K)
+            sage: H = L(a*x + y -1, x + a*y + 1, x - 1, y - 1)
+            sage: H.affine_fundamental_group()
+            Traceback (most recent call last):
+            ...
+            TypeError: the base field is not in QQbar
+            sage: L.<t> = OrderedHyperplaneArrangements(QQ)
+            sage: L([t - j for j in range(4)]).affine_fundamental_group()
+            Finitely presented group < x0, x1, x2, x3 |  >
+            sage: L.<x, y, z> = OrderedHyperplaneArrangements(QQ)
+            sage: L(L.gens() + (x + y + z + 1,)).affine_fundamental_group().sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3 | x3^-1*x2^-1*x3*x2, x3^-1*x1^-1*x3*x1,
+                               x3^-1*x0^-1*x3*x0, x2^-1*x1^-1*x2*x1,
+                               x2^-1*x0^-1*x2*x0, x1^-1*x0^-1*x1*x0 >
+            sage: A = OrderedHyperplaneArrangements(QQ, names=())
+            sage: H = A(); H
+            Empty hyperplane arrangement of dimension 0
+            sage: H.affine_fundamental_group()
+            Finitely presented group <  |  >
+        """
+        K = self.base_ring()
+        if not K.is_subring(QQbar):
+            raise TypeError('the base field is not in QQbar')
+        if self._affine_fundamental_group:
+            return self._affine_fundamental_group
+        n = self.dimension()
+        r = len(self)
+        if n == 0:
+            return FreeGroup(0) / []
+        if n == 1:
+            G = FreeGroup(r) / []
+            dic = {j: G.gen(j) for j in range(r)}
+            dic[r] = [prod(G.gens()) ** -1]
+            self._affine_fundamental_group = G
+            self._affine_meridians = dic
+            return G
+        if n == 2:
+            S = self.parent().ambient_space().symmetric_space()
+            coord = vector((1,) + S.gens())
+            Af = AffinePlaneCurveArrangements(K,
+                                              names=self.parent().variable_names())
```

Comment:
```suggestion
            Af = AffinePlaneCurveArrangements(K, names=self.parent().variable_names())
```
The line break only "saves" two characters at the cost of readability.